src/audio/module_adapter/module_adapter.c

```diff
@@ -98,6 +98,84 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 	return NULL;
 }
 
+
+/**
+ * \brief Create a codec adapter component for loadable module.
+ * \param[in] drv - component driver pointer.
+ * \param[in] config - component ipc descriptor pointer.
+ * \param[in] module_interface - component module interface.
+ * \param[in] spec - module proprietary configuration.
+ * \param[in] spec_size - size of  module proprietary configuration.
+ *
+ * \return: a pointer to newly created codec adapter component.
+ */
+struct comp_dev *module_adapter_new_loadable(const struct comp_driver *drv,
```

Comment:
this whole function can also be replaced if we include the size in the *spec data

Reply:
Yes, but it still require modifications in all SOF modules implementations including build in ones (aria, .ascr, src etc.).  I do not insist to use ops.create2() since I don't like that at all. However modifying ops.create() as well as switching to structure you proposed above (struct ipc_specific_config) will involve touching many components. I think we need to find out the best solution here.

Reply:
Fixed using **struct ipc_config_process**.